server/cluster.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"database/sql"
+
+	"github.com/mattermost/squirrel"
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost/server/public/model"
+)
+
+func pingClusterDiscoveryTable(db *sql.DB, driverName, clusterName string) ([]*model.ClusterDiscovery, error) {
+	var phf squirrel.PlaceholderFormat
+	phf = squirrel.Question
+	if driverName == model.DatabaseDriverPostgres {
+		phf = squirrel.Dollar
+	}
+
+	builder := squirrel.StatementBuilder.PlaceholderFormat(phf)
+
+	query := builder.Select("Id,Hostname").From("ClusterDiscovery").
+		Where(squirrel.Eq{"Type": model.CDSTypeApp, "ClusterName": clusterName}).
+		Where(squirrel.Gt{"LastPingAt": model.GetMillis() - model.CDSOfflineAfterMillis})
```

Comment:
My suggestion was to order the results by hostname from the DB itself so that we don't have to do `sliceToMap` again to see what changed. We can simply iterate the old and new slices and break out if any element differs.